os/kernel/semaphore/sem_post.c

```diff
@@ -201,7 +201,7 @@ int sem_post(FAR sem_t *sem)
 	irqstate_t saved_state;
 	int ret = ERROR;
 	size_t caller_retaddr = 0;
-	ARCH_GET_RET_ADDRESS(caller_retaddr);
+	caller_retaddr = (mmaddress_t)__builtin_return_address(0);
```

Comment:
1. Do we need to use the local variable? I think we can use the function at line 217 directly without it.
2. If we need it indeed, I think we can assign the return value at definition at line 203. I mean no need to assign to 0 and the function return again.